modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -762,57 +833,88 @@ sub apply_tweaks {
+                    $tweakStructure->{Return}->{Field} = $attrib_name;
                     my $old_value = stringify($analysis->$attrib_name());
-
+                    $tweakStructure->{Return}->{OldValue} = $old_value;
                     if($operator eq '?') {
-                        print "Tweak.Show    \tanalysis[$analysis_name].$attrib_name ::\t$old_value\n";
+                        $tweakStructure->{Action} = "SHOW";
+                        $tweakStructure->{Return}->{NewValue} = $tweakStructure->{Return}->{OldValue};
+                        push @response, "Tweak.Show  \tanalysis[$analysis_name].$attrib_name ::\t$old_value\n";
                     } elsif($operator eq '#') {
-                        print "Tweak.Error   \tDeleting of Analysis attributes is not supported\n";
+                        my $errorMsg = "Tweak.Error  \tDeleting of Analysis attributes is not supported\n";
+                        $tweakStructure->{Action} = "DELETE";
+                        $tweakStructure->{Error} = $errorMsg;
+                        push @response, $errorMsg;
                     } else {
-                        print "Tweak.Changing\tanalysis[$analysis_name].$attrib_name ::\t$old_value --> ".stringify($new_value)."\n";
-
+                        $tweakStructure->{Action} = "SET";
+                        $tweakStructure->{Return}->{NewValue} = stringify($new_value);
+                        push @response, "Tweak.Changing  \tanalysis[$analysis_name].$attrib_name ::\t$old_value --> ".stringify($new_value)."\n";
                         $analysis->$attrib_name( $new_value );
                         $need_write = 1;
                     }
                 } else {
-                    print "Tweak.Error   \tAnalysis does not support '$attrib_name' attribute\n";
+                    my $errorMsg = "Tweak.Error  \tAnalysis does not support '$attrib_name' attribute\n";
+                    $tweakStructure->{Error} = $errorMsg;
+                    push @response, $errorMsg;
                 }
+
+                push @{$responseStructure->{Tweaks}}, $tweakStructure;
             }
 
         } elsif($tweak=~/^resource_class\[([^\]]+)\]\.(\w+)(\?|=(.+))$/) {
+            my $tweakStructure;
             my ($rc_pattern, $meadow_type, $operator, $new_value_str) = ($1, $2, $3, $4);
-
+            $tweakStructure->{Object}->{Type} = "Resource class";
             my $resource_classes = $self->collection_of( 'ResourceClass' )->find_all_by_pattern( $rc_pattern );
-            print "Tweak.Found   \t".scalar(@$resource_classes)." resource_classes matching the pattern '$rc_pattern'\n";
+            push @response, "Tweak.Found  \t".scalar(@$resource_classes)." resource_classes matching the pattern '$rc_pattern'\n";
 
             if($operator eq '?') {
                 foreach my $rc (@$resource_classes) {
                     my $rc_name = $rc->name;
+                    $tweakStructure->{Object}->{Id} = $rc->dbID;
```

Comment:
`$tweakStructure` needs to be reset every loop otherwise the `Tweaks` list will contain references to the same object